docs/compass/10-01-rate-limiting.md

```diff
@@ -0,0 +1,28 @@
+# CMP Rate Limiting
+
+Rate-limiting is implemented in CMP using the [Istio's local rate-limiting](https://istio.io/latest/docs/tasks/policy-enforcement/rate-limit/#local-rate-limit) functionality which provides enabling rate-limiting at instance level. In Kubernetes terms, that can be thought of as rate-limiting per Deployment. Enablement is done by applying an [EnvoyFilter](https://istio.io/latest/docs/reference/config/networking/envoy-filter/) which is responsible for maintaining **token-buckets**. A token-bucket is a simple data structure which contains a fixed amount of tokens. For each HTTP request that arrives a token is removed from the token-bucket. In case the token-bucket gets emptied, the next incoming request gets rate-limited causing an early termination of the request handling and returning an HTTP status-code of 429 (Too Many Requests). The token bucket gets refilled on regular intervals with a fixed amount of tokens. The initial token count, refill rate and the amount of tokens per refill can be configured in the EnvoyFilter's resource description.
```

Comment:
"rate-limiting per Deployment" isn't it per Pod of a given deployment?

Reply:
Correct. I tested this and I saw that the more the replicas were, the later we started rate-limiting, so yes it is per Pod. I changed it.